### DISCUSSION TITLE
Does NetTopologySuite calculate the perpendicular distance from a LineString and a Point?

### DISCUSSION THREAD

**Discussion by ranouf** [NetTopologySuite/NetTopologySuite#599] [Q&A]
Hi,

Since .Net Core 2.2, I'm now able to use your library, thanks, it s really usefull.

I have a question about the **distance calculation method between a LineString and a Point**. 

I visited the documentation available here: https://nettopologysuite.github.io/html/class_net_topology_suite_1_1_operation_1_1_distance_1_1_distance_op.html#a511116a82b4f03dbae149a7eef9d5235
But after reading the explanation, I was still not sure.

Does your library calculate it perpendicularly, like this?

![image](https://user-images.githubusercontent.com/1764519/51636078-57934b80-1f26-11e9-9175-dbdc4b1d8d8e.png)

Or does it take the middle of the LineString to calculate the distance? Or another type of calculation?

Thanks for your explanations :)

**Comment by FObermaier** [comment]
For `POINT` - `LINESTRING` distances the result is the perpendicular distance to the closest LineSegment:

```C#
var pt = new Point(0, 1);
var ls = new LineString(new[] {
        new Coordinate(-1.4, 2), 
        // this is the closest segment to POINT (0 1)
        new Coordinate(-1, -1), new Coordinate(1, 1), 
        new Coordinate(1.2, 3)
});
double distance = DistanceOp.Distance(ls, pt);
Assert.That(distance, Is.EqualTo(Math.Sqrt(2) * 0.5).Within(0.00001));
```

The computation is performed here:
https://github.com/NetTopologySuite/NetTopologySuite/blob/583ae18b593e5835630aa7e67bb82dd8eb349ccd/NetTopologySuite/Algorithm/Distance.cs#L147-L198

**Comment by ranouf** [comment]
Thanks @FObermaier !

**Comment by ranouf** [comment]
![image](https://user-images.githubusercontent.com/1764519/52078854-41a70b80-2562-11e9-87cb-462a9c1b166a.png)

Hi,

What happens in this case? the point is not perpendicular to the line. 

Case 1:
![image](https://user-images.githubusercontent.com/1764519/52078922-6ef3b980-2562-11e9-8529-60f197c36f18.png)

The line is "increased" and the we calculate the distance perpendicularly?

![image](https://user-images.githubusercontent.com/1764519/52078996-9a76a400-2562-11e9-8918-d52b7aee7923.png)

Or the calculation is made to the closest end of the line to the point?

**Comment by FObermaier** [comment]
@ranouf , in that case it computes the distance to the closest endpoint:
```C#
var pt = new Point(0, 1);
var ls = new LineString(new[] {
        new Coordinate(-1.4, 2), 
        // this is the closest segment to POINT (0 1)
        new Coordinate(-1, -1), new Coordinate(1, 1), 
        new Coordinate(1.2, 3)
});
double distance = DistanceOp.Distance(ls, pt);
Assert.That(distance, Is.EqualTo(Math.Sqrt(2) * 0.5).Within(0.00001));

pt = new Point(-2, 2);
distance = pt.Distance(ls);
Assert.That(distance, Is.EqualTo(0.6).Within(0.00001));

pt = new Point(-2, -2);
distance = pt.Distance(ls);
Assert.That(distance, Is.EqualTo(Math.Sqrt(2)).Within(0.00001));
```

**Comment by ranouf** [comment]
Thank you!
